src/main/kotlin/com/github/gradle/android/i18n/import/XlsImporter.kt

```diff
@@ -1,64 +1,217 @@
 package com.github.gradle.android.i18n.import
 
+import com.github.gradle.android.i18n.conf.Configuration
+import com.github.gradle.android.i18n.model.*
 import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.ss.usermodel.Workbook
 import org.apache.poi.ss.usermodel.WorkbookFactory
 import org.gradle.api.Project
+import java.io.File
 import java.io.InputStream
+import java.nio.file.Path
+import java.nio.file.Paths
 
 /**
  * Android i18n resources importer from `.xls` and `.xlsx` sources.
  */
-class XlsImporter(project: Project) : AbstractImporter(project) {
-
-    override fun generate(inputStream: InputStream, config: ImportConfig, handler: ImportHandler) {
-
-        /**
-         * Map storing each locale to its corresponding column index.
-         */
-        val localizedColumns = mutableMapOf<Int, String>()
-
-        readInput(inputStream, config) { row, readLocales ->
-            if (readLocales && row.rowNum == 0) {
-                // Reading locales row and initializing `StringResources`.
-                row.cellIterator()
-                        .asSequence()
-                        .filter { it.columnIndex > 0 }
-                        .filter { it.stringCellValue.isNotBlank() }
-                        .forEach { cell ->
-                            val locale = cell.stringCellValue
-                            handler.addLocale(locale)
-                            localizedColumns[cell.columnIndex] = locale
-                        }
-            } else if (row.rowNum > 0) {
-                // Reading other data row.
-                val key = row.getCell(0)?.stringCellValue
-
-                val translations = localizedColumns
-                        .map { Pair(it.value, row.getCell(it.key)?.stringCellValue) }
-                        .toMap()
-
-                handler.addTranslations(key, translations)
-            }
+class XlsImporter(private val project: Project) : AbstractImporter(project) {
+
+    override fun generate(inputStream: InputStream, config: ImportConfig) {
+
+        val workbook = WorkbookFactory.create(inputStream) // TODO .use
+
+        val projectData = if (project.isSingleModule()) {
+            workbook.toSingleModuleProjectData(config)
+        } else {
+            workbook.toMultiModuleProjectData(config)
```

Comment:
Why code need to be different if multimodule or not ?

Reply:
Because the same input Excel file can be imported into a single module Gradle project or a multi module project, and the result will be different:
- For a single module project, all the sheets will be considered as one single sheet and imported into the `strings.xml` files of that single module
- For a multi module project, each sheet will be imported into the `strings.xml` files of a submodule

It means that the content of the input file is not enough to decide of the number and structure of the `strings.xml` files to create. 
The plugin needs to know if the output is a multi module project to put the files in the appropriate subfolders.

Reply:
To do: add documentation in the README